.github/workflows/lint-testdata.yml

```diff
@@ -0,0 +1,43 @@
+name: Testdata Lint Check
+
+# Trigger the workflow on pull requests and direct pushes to any branch
+on:
+  push:
+  pull_request:
+
+jobs:
+  check-lint-for-project-v4:
+    uses: kubernetes-sigs/kubebuilder/.github/workflows/lint-testdata-template.yml@master
```

Comment:
So I think using a template is a really neat approach, but not sure it is necessary. Since we already have a target being added to the Makefile that *should* automatically capture any new testdata directories I think it would be easier to maintain an action workflow that simply calls that make target. 

That being said, if there is agreement from other maintainers on taking this approach I am fine with it as well.